Src/Base/AMReX_Geometry.H

```diff
@@ -275,12 +276,24 @@ public:
         }
     }
 
+    bool outsideRoundoffDomain (AMREX_D_DECL(Real x, Real y, Real z)) const;
+    bool insideRoundoffDomain (AMREX_D_DECL(Real x, Real y, Real z)) const;
+
+    void computeRoundoffDomain ();
+
 private:
     void read_params ();
 
     // is_periodic and RealBox used to be static
     bool    is_periodic[AMREX_SPACEDIM] = {AMREX_D_DECL(false,false,false)};
     RealBox prob_domain;
+
+    // Due to round-off errors, not all floating point numbers for which plo >= x < phi
+    // will map to a cell that is inside "domain". "roundoff_domain" stores a phi
+    // that is very close to that in prob_domain, and for which all floating point numbers
+    // inside it according to a naive inequality check will map to a cell inside domain.
+    RealBox roundoff_domain;
```

Comment:
I wonder if this should be two arrays of partice_reals.

Reply:
Or should we use `Real` everywhere without using `ParticleReal` at all?  Then when particle codes will call `outsideRoundoffDomain` that takes `Real`, we let implicit conversion happen if the types are different.

Reply:
The latter seems to work in all the cases I've been testing, anyway.